final_task/rss_reader/argparse_handler.py

```diff
@@ -0,0 +1,160 @@
+import argparse
+import datetime
+import logging
+import os
+import sys
+import urllib.request
+from typing import Optional
+
+import custom_error
+
+VERSION = 5.0
+
+
+def check_if_date_in_arguments() -> bool:
+    """Checks if date is arguments"""
+    return '--date' in sys.argv
+
+
+def check_if_help_or_version_in_arguments() -> Optional[str]:
+    """Checks if help or version in arguments"""
+    if '--help' in sys.argv or '-h' in sys.argv:
+        return 'help'
+    elif '--version' in sys.argv:
+        return 'version'
+    else:
+        return None
+
+
+def check_the_arguments_amount() -> None:
+    """Checks if there 1 or 0 arguments and raises exception"""
+    if len(sys.argv) < 2:
+        raise custom_error.NotEnoughArgumentsError
+
+
+def link_found() -> bool:
+    """Checks if link in arguments"""
+    for index, value in enumerate(sys.argv):
+        if "http" in value:
+            if value.count("'") == 2:
+                value = value[1:-1]
+                sys.argv[index] = value
+            if index != 1:
+                sys.argv.insert(1, sys.argv.pop(index))
+            return True
+    return False
+
+
+def check_if_url_or_date_in_arguments() -> str:
+    """Checks if url in args and puts it to sys.argv[1] else raises exception"""
+    flag = check_if_help_or_version_in_arguments()
+    if not flag:
+        if not check_if_date_in_arguments():
+            if link_found():
+                return 'link_only'
+            else:
+                raise custom_error.UrlNotFoundInArgsError
+        else:
+            if link_found():
+                return 'link_and_date'
+            else:
+                return 'date_only'
+    else:
+        return flag
+
+
+def check_the_connection(rss_url: str) -> str:
+    """Checks the connection"""
+    try:
+        urllib.request.urlopen(rss_url)
+        return 'ok'
+    except urllib.request.HTTPError as e:
+        return f'{e.code}: {e.reason}'
+    except urllib.request.URLError as e:
+        return f'{e.reason}'
+
+
+def check_if_url_is_valid(rss_url: str) -> bool:
+    """Checks if link is valid"""
+    result = urllib.request.urlparse(rss_url)
+    return all([result.scheme, result.netloc, "." in result.netloc, len(result.netloc) > 2])
+
+
+def valid_url(rss_url: str) -> str:
+    """Checks if url is valid and connection successful"""
+    if check_if_url_is_valid(rss_url):
+        logging.info('Valid url')
+
+        connection = check_the_connection(rss_url)
+        if connection != 'ok':
+            raise custom_error.ConnectionFailedError(f"Connection failed: {connection}")
+
+        logging.info('Connection successful')
+        return rss_url
+    else:
+        msg = f"Not a valid URL value: '{rss_url}'"
+        raise custom_error.NotValidUrlError(msg)
+
+
+def valid_limit(limit: Optional[str]) -> Optional[int]:
+    """Checks if limit is valid"""
+    try:
+        limit = int(limit)
+        if limit > 0:
+            return limit
```

Comment:
Same here, your function name says, that limit either valid or not, so bool fits here as well

Reply:
This function is used here `parser.add_argument("--limit", type=valid_limit, default=None, help="Limit news topics if this parameter provided")` to convert limit to int and check if it's correct at the same time, i can rework docstring of this function to be more informative